src/commands/org/create/sandbox.ts

```diff
@@ -155,9 +173,13 @@ export default class CreateSandbox extends SandboxCommandBase<SandboxCommandResp
       );
       delete sandboxReq.ActivationUserGroupName;
     }
+
     return {
       ...sandboxReq,
-      ...(this.flags.clone ? { SourceId: await this.getSourceId() } : {}),
+      ...(srcSandboxName ?? this.flags['source-sandbox-name']
```

Comment:
no need to check for `--source-sandbox-name` here, `srcSandboxName` should have that value (`requestFunctions.createSandboxRequest` takes these flags as `requestOptions` and merges them with the def.file).